packages/smooth_app/lib/pages/scan/scanner_overlay.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:smooth_app/data_models/continuous_scan_model.dart';
+import 'package:smooth_app/pages/scan/scan_page_helper.dart';
+import 'package:smooth_app/widgets/smooth_product_carousel.dart';
+import 'package:smooth_ui_library/animations/smooth_reveal_animation.dart';
+import 'package:smooth_ui_library/widgets/smooth_view_finder.dart';
+import 'package:visibility_detector/visibility_detector.dart';
+
+class ScannerOverlay extends StatelessWidget {
+  const ScannerOverlay({
+    required this.scannerWidget,
+    required this.model,
+    required this.restartCamera,
+    required this.stopCamera,
+  });
+
+  final Widget scannerWidget;
+  final ContinuousScanModel model;
+  final Function() restartCamera;
+  final Function() stopCamera;
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (
+        BuildContext context,
+        BoxConstraints constraints,
+      ) {
+        final Size screenSize = MediaQuery.of(context).size;
+        final Size scannerSize = Size(
+          screenSize.width * 0.6,
+          screenSize.width * 0.33,
+        );
+        final double carouselHeight =
+            constraints.maxHeight / 1.81; // roughly 55% of the available height
```

Comment:
Strange to see this `1.81` again, after `continuous_scan_page.dart`.
I don't know what if refers to. I suggest to use a constant instead.
And maybe to use `0.55` instead of `1.81`, that would save the comment `// roughly 55% of the available height`.